### ISSUE TITLE
workflow.setAttackRoll() might break if the target has Reactions and you set the AttackRoll to a NumericTerm.

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#1140]
In GitLab by @thatlonelybugbear on Jan 27, 2023, 12:58

_Just tested on MidiQOL 10.0.27, Dnd5e 2.1.4._

So I am using an Actor onUse macro to modify the `AttackRoll` during `preCheckHits`, depending on the result of the normal Attack Roll.

Relevant part of the macro
```js
let attackTotal = workflow.attackTotal;
let addRolltoAttack = await new Roll('1d6').evaluate();
let changed = Number(attackTotal) + Number(addRolltoAttack.total);
let newAttackRoll = await new Roll(`${changed}`).evaluate();
this.setAttackRoll(newAttackRoll);
```
As a result the new "Roll" will not have `workflow.attackRoll.terms[0].results[0].result` as it is a NumericTerm and has only `workflow.attackRoll.terms[0].total` instead.

With that in mind, when the target has available Reactions and `configSettings.showReactionAttackRoll` is set to `d20`, https://gitlab.com/tposney/midi-qol/-/blob/v10/src/module/utils.ts#L2851 will error out when reading `results` as this will be undefined.

I made a small change in that line locally to test it and it seems to resolve the issue
```js
const theRoll = attackRoll?.terms[0]?.results ? attackRoll.terms[0].results[0].result : attackRoll?.terms[0]?.total ? attackRoll.terms[0].total : "";
```

It all boils down to the `newAttackRoll` not being an actual dice roll but a "Number roll", so I am not sure if that is the best way to move forward.

PS: The above macro was working fine when I first created it for a user on v9, but many things have changed in between.

**Comment by tposney** [comment]
Did not really consider that. Implicitly I assume an attack roll is a d20 roll - since that's all that dnd5e can generate - but did not consider it being changed by a macro.

I'll include the suggested fix in 10.0.28

**Comment by tposney** [comment]
In GitLab by @thatlonelybugbear on Feb 4, 2023, 01:26

Yeah true d20, not sure why I included d6 :sweat_smile:
